### DISCUSSION TITLE
Pipeline error

### DISCUSSION THREAD

**Discussion by raisangeeta11** [actions/virtual-environments#6556] [Installed software]
Getting the error when running the pipeline using self hosted agent in Azure

Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:testCompile (default-testCompile) on project PersonaRegression: Fatal error compiling: invalid target release: 17 -> [Help 1]

Copy of my YAML file
# Maven
# Build your Java project and run tests with Apache Maven.
# Add steps that analyze code, save build artifacts, deploy, and more:
# https://docs.microsoft.com/azure/devops/pipelines/languages/java

trigger:
- develop

pool:
  vmImage: 'windows-latest'

steps:
- task: MavenAuthenticate@0
  inputs:
    artifactsFeeds: 'azure.testautomation'

- task: Maven@3
  inputs:
    mavenPomFile: 'pom.xml'
    mavenOptions: '-Xmx3072m'
    javaHomeOption: 'JDKVersion'
    jdkVersionOption: '1.8'
    jdkArchitectureOption: 'x64'
    publishJUnitResults: true
    testResultsFiles: '**/surefire-reports/TEST-*.xml'
    goals: 'package'

I am not sure if it has got anything to do with specific versions set in the pom.xml or something not installed on the Microsoft hosted agent box
  <properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    <maven.compiler.source>1.8</maven.compiler.source>
    <maven.compiler.target>1.8</maven.compiler.target>
    <cucumber.version>6.9.0</cucumber.version>
    <maven.compiler.version>3.8.1</maven.compiler.version>
    <maven.surefire.version>3.0.0-M5</maven.surefire.version>
    <java.version>17</java.version>
    <junit.version>4.13.2</junit.version>
    <testng.version>7.6.1</testng.version>
    <selenium.version>4.5.3</selenium.version>
    <webdrivermanager.version>5.3.0</webdrivermanager.version>
    <cucumber.version>7.8.1</cucumber.version>
  </properties>

**Comment by al-cheb** [comment]
For general questions about using the runner images or writing your Actions workflow, please open requests in the [GitHub Actions Community Forum](https://github.community/c/github-actions/41).
